.github/workflows/_snap-release.yaml

```diff
@@ -0,0 +1,39 @@
+on:
+  workflow_call:
+    inputs:
+      track:
+        description: |
+          The track to which charm should be released. By default, the latest track
+          will be used.
+        default: latest
+        required: false
+        type: string
+      risk-level:
+        description: |
+          The risk level of the channel to which charm should be released. Valid
+          inputs are: edge, beta, candidate, stable. By default, the snap will be
+          released to a channel with "edge" risk level. 
+        default: edge
+        required: false
+        type: string
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: Publish Snap
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: snap-artifact
+      - id: get-snap-name
+        run: echo "SNAP_FILE=$(ls *.snap)" >> $GITHUB_OUTPUT
+      - name: Release snap to the specified channel
+        uses: snapcore/action-publish@v1
+        env:
+          SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
+        with:
+          snap: ${{ steps.get-snap-name.outputs.SNAP_FILE }}
+          release: ${{ inputs.track }}/${{ inputs.risk-level }}
```

Comment:
I thorough that we can provide multiple tracks in inputs, so we would need some for loop here or maybe action-publish supports multiple tracks here?

Reply:
I was planning to add this feature as a future improvement, but you are right, it is better to include it here. I have pushed a new commit enabling multi-channel release support.